### PR TITLE
Update pandas.util.testing to pandas.testing

### LINKED ISSUE #2539: CLN: remove deprecated test usages
>  /home/runner/work/ibis/ibis/ibis/tests/all/test_temporal.py:5: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.

IOW switch pandas.util.testing -> pandas.testing.

### PR DESCRIPTION
**Pull request by matthewmturner**
Closes #2539

### CODE REVIEW

File: ibis/backends/impala/tests/test_pandas_interop.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import numpy as np
 import pandas as pd
 import pytest
-from pandas.util.testing import assert_frame_equal
```
Comment (jreback) [inline ibis/backends/impala/tests/test_pandas_interop.py:4]: can you change these to
`import pandas.testing  as tm` to be consitent

File: ibis/backends/pandas/tests/execution/test_cast.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import decimal
 
 import pandas as pd
-import pandas.util.testing as tm  # noqa: E402
```
Comment (jreback) [inline ibis/backends/pandas/tests/execution/test_cast.py:4]: if you can remove the noqa (if its not needed then remove the import)

File: ibis/backends/sqlite/tests/test_functions.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 import numpy as np
 import pandas as pd
-import pandas.util.testing as tm  # noqa: E402
```
Comment (jreback) [inline ibis/backends/sqlite/tests/test_functions.py:9]: same for all

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by datapythonista — APPROVED** [review]
Great cleanup @matthewmturner, thanks for working on this.

@jreback all your comments are addressed, if you want to have another look.

**Comment by datapythonista** [comment]
@matthewmturner there are couple of errors in the CI, not sure why those functions are not in `pandas.testing` anymore, can you have a look please?

**Comment by matthewmturner** [comment]
could it be related to #2538 which is now raising test failures compared to my first commit when these tests passed?

I also only see the following 4 functions in `pandas.testing` from both documentation and checking directly.

```
In [1]: import pandas.testing as tm

In [2]: dir(tm)
Out[2]: 
['__all__',
 '__builtins__',
 '__cached__',
 '__doc__',
 '__file__',
 '__loader__',
 '__name__',
 '__package__',
 '__spec__',
 'assert_extension_array_equal',
 'assert_frame_equal',
 'assert_index_equal',
 'assert_series_equal']
```

Is it expected that all functionality in  `pandas.util.testing` is in `pandas.testing`?

**Comment by matthewmturner** [comment]
@datapythonista any thoughts here? Im happy to switch back to `pandas.util.testing` for the `makeDataframe` functionality thats missing in `pandas.testing`?  I had thought all functionality in `pandas.util.testing` would be `pandas.testing` but it appears not.

Seems like an easy fix just wanted to make sure it aligned to your expectations.

**Comment by datapythonista** [comment]
I think `makeDataFrame` is not part of the pandas public API, and not supposed to be used outside of pandas, that's why it's not in pandas.testing. I think the "right" solution is to avoid using it and create the dataframe manually. Does this sound right @jreback ?

**Comment by jreback** [comment]
yes - just create a test dataframe here instead

**Comment by matthewmturner** [comment]
@jreback updated to manually creating test df. let me know if anything else.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @matthewmturner nice! keep em coming!

**Comment by matthewmturner** [comment]
@jreback are there any issues you want prioritized for next release?

**Comment by datapythonista** [comment]
> @jreback are there any issues you want prioritized for next release?

#2533 would be a really good one to have fixed

**Comment by matthewmturner** [comment]
> 
> 
> > @jreback are there any issues you want prioritized for next release?
> 
> #2533 would be a really good one to have fixed

Ok - will give it a look.

**Pull request merged**